LanguageFeatures/Patterns/type_inference_A16_t01.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Calculate the static type of the pattern. Using that value type,
+/// recurse through the pattern again downwards to the leaf subpatterns filling
+/// in any missing types in the pattern. This process may also insert implicit
+/// coercions and casts from dynamic when values flow into a pattern during
+/// matching.
+///
+/// @description Check a static type of a parenthesized pattern. Test that
+/// missing types in a type schema are filled from the initialising expression
+/// and implicit coercions and casts from dynamic are performed
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "../../Utils/static_type_helper.dart";
+import "../../Utils/expect.dart";
+
+main() {
+  var <double>[v1] = [42];
```

Comment:
Yes, this is good! We need to cover the case where some elements in a pattern are filled in based on other parts of the same pattern, not just the cases where the initializing expression runs the whole show.